docbrown/db/src/graphgen/preferential_attachment.rs

```diff
@@ -0,0 +1,93 @@
+use crate::graph::Graph;
+use rand::prelude::*;
+use docbrown_core::Direction;
+use std::collections::HashSet;
+
+//let mut weights:Vec<usize> =view.vertices().map(|v|v.degree()).collect();
+
+pub fn preferential_attachment(graph:&Graph,vertices_to_add:usize,edges_per_step:usize){
+
+    let mut rng = rand::thread_rng();
+    let mut latest_time = graph.latest_time();
+
+    let view = graph.window(i64::MIN,i64::MAX);
+    let mut ids:Vec<u64> = view.vertex_ids().collect();
+    let mut degrees:Vec<usize> = view.vertices().map(|v| v.degree()).collect();
+    let mut edge_count:usize = degrees.iter().sum();
+
+    let mut max_id = match ids.iter().max() {
+        Some(id) => {*id},
+        None=>0
+    };
+
+    if(ids.len()==0){
+        graph.add_vertex(latest_time,0,&vec![]);
+        degrees.push(0);
+        ids.push(0);
+    }
+    while ids.len() < edges_per_step {
+        max_id+=1;
+        latest_time+=1;
+        let prev_node = ids[ids.len()-1];
+        graph.add_vertex(latest_time,max_id,&vec![]);
+        graph.add_edge(latest_time,max_id,prev_node,&vec![]);
```

Comment:
Shouldn't we add the edge in both directions?

Reply:
May thinking with this is that with the undirected views etc a user wanting the edge in both ways could just utilise that.